civis_jupyter_notebooks/platform_persistence.py

```diff
@@ -45,23 +47,45 @@ def initialize_notebook_from_platform(notebook_path):
     logger.info('Notebook file ready')
 
     if hasattr(notebook_model, 'requirements_url') and notebook_model.requirements_url:
-        __pull_and_load_requirements(notebook_model.requirements_url)
+        __pull_and_load_requirements(notebook_model.requirements_url, notebook_path)
 
 
-def __pull_and_load_requirements(url):
+def __pull_and_load_requirements(url, notebook_path):
     logger.info('Pulling down the requirements file')
     r = requests.get(url)
 
     if r.status_code != 200:
         raise NotebookManagementError('Failed to pull down requirements.txt file from S3')
 
     logger.info('Writing contents of requirements file')
-    with open('requirements.txt', 'wb') as requirements:
+    requirements_path = os.path.join(os.path.dirname(notebook_path), 'requirements.txt')
+    with open(requirements_path, 'wb') as requirements:
         requirements.write(r.content)
 
     logger.info('Requirements file ready')
 
 
+def install_requirements(requirements_path):
+    while os.path.isdir(requirements_path):
+        requirements_file = os.path.join(requirements_path, 'requirements.txt')
+        logger.info('Looking for requirements at %s' % requirements_file)
+        if not os.path.isfile(requirements_file):
+            requirements_path = os.path.dirname(requirements_path)
+            continue
+
+        logger.info('Installing packages from %s' % requirements_file)
+        try:
+            subprocess.check_output(
+                    [sys.executable, '-m', 'pip', 'install', '-r', requirements_file],
+                    stderr=subprocess.STDOUT
+                    )
+            logger.info('requirements.txt installed')
+            break
```

Comment:
why the `break`?

Reply:
Because if we find a requirements.txt again, we don't want the `while` loop to keep going up the tree

Reply:
Gotcha - didn't see the while loop. do you want to have a base case in place that cuts climbing at `/root/work`? Right now the code climbs past where the git repo is.

Reply:
I dunno, I'm inclined to keep it. I guess what would a user expect if they used a custom image with a `requirements.txt` in `/root`?

Reply:
That's a silly example...I guess in a custom image they would just have installed those requirements already. I can check for `/root` and stop there